server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/NonSnarlIdpMetadataManager.java

```diff
@@ -213,10 +213,10 @@ protected String getSpName(ExtendedMetadataDelegate provider) throws MetadataPro
     protected String getHostedSpName(ExtendedMetadataDelegate provider) throws MetadataProviderException {
         String key = getSpName(provider);
         ExtendedMetadata extendedMetadata = getExtendedMetadata(key, provider);
-        if (extendedMetadata.isLocal()) {
+        if (extendedMetadata != null && extendedMetadata.isLocal()) {
             return key;
         } else {
-            return null;
+            throw new MetadataProviderException("Extended metadata not found for provider [" + provider + "]");
```

Comment:
Better to use key (aka service provider name) to identify provider here

Reply:
Will do